core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -1759,6 +1759,9 @@ protected SqlSelect createSourceSelectForDelete(SqlDelete call) {
   @Override public RelDataType getValidatedNodeType(SqlNode node) {
     RelDataType type = getValidatedNodeTypeIfKnown(node);
     if (type == null) {
+      if (node.isA(ImmutableSet.of(SqlKind.IDENTIFIER))) {
+        throw newValidationError(node, RESOURCE.unknownIdentifier(node.toString()));
```

Comment:
How about using `node.getKind() == SqlKind.IDENTIFIER` ?

Reply:
Done.

I forced push the commit since this is a small change whose history probably should not be kept. Curious what the recommended approach is for ammending PRs in calcite.